### PR TITLE
[FLINK-18839][docs] Add documentation about how to use catalog in Python Table API

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request add documentation about how to use catalog in Python Table API*

## Verifying this change

This change is a documentation rework without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: docs/dev/table/catalogs.md [hunk 1]
```diff
@@ -212,6 +233,59 @@ catalog.createTable(
...
 {% endhighlight %}
+</div>
+<div data-lang="Python" markdown="1">
+{% highlight python %}
+from pyflink.table import *
+from pyflink.table.catalog import HiveCatalog, CatalogDatabase, ObjectPath, CatalogBaseTable
+from pyflink.table.descriptors import Kafka
+
+settings = EnvironmentSettings.new_instance().in_batch_mode().use_blink_planner().build()
+t_env = StreamTableEnvironment.create(environment_settings=settings)
```
Comment (WeiZhong94) [inline docs/dev/table/catalogs.md:244]: BatchTableEnvironment?

File: docs/dev/table/connectors/jdbc.md [hunk 2]
```diff
@@ -375,6 +375,38 @@ tableEnv.registerCatalog("mypg", catalog)
...
+    """
+    def __init__(self, catalog_name, default_database, username, pwd, base_url):
+        from pyflink.java_gateway import get_gateway
+        gateway = get_gateway()
+
+        j_jdbc_catalog = gateway.jvm.org.apache.flink.connector.jdbc.catalog.JdbcCatalog(
+            catalog_name, default_database, username, pwd, base_url)
+        super(JdbcCatalog, self).__init__(j_jdbc_catalog)
+
+environment_settings=EnvironmentSettings.new_instance().in_streaming_mode().use_blink_planner().build()
```
Comment (WeiZhong94) [inline docs/dev/table/connectors/jdbc.md:394]: add space before and after the '='?

File: docs/dev/table/connectors/jdbc.zh.md [hunk 3]
```diff
@@ -375,6 +375,38 @@ tableEnv.registerCatalog("mypg", catalog)
...
+    """
+    def __init__(self, catalog_name, default_database, username, pwd, base_url):
+        from pyflink.java_gateway import get_gateway
+        gateway = get_gateway()
+
+        j_jdbc_catalog = gateway.jvm.org.apache.flink.connector.jdbc.catalog.JdbcCatalog(
+            catalog_name, default_database, username, pwd, base_url)
+        super(JdbcCatalog, self).__init__(j_jdbc_catalog)
+
+environment_settings=EnvironmentSettings.new_instance().in_streaming_mode().use_blink_planner().build()
```
Comment (WeiZhong94) [inline docs/dev/table/connectors/jdbc.zh.md:394]: ditto

File: docs/dev/table/hive/hive_dialect.zh.md [hunk 4]
```diff
@@ -80,6 +82,22 @@ tableEnv.getConfig().setSqlDialect(SqlDialect.HIVE);
 tableEnv.getConfig().setSqlDialect(SqlDialect.DEFAULT);
 
 {% endhighlight %}
+</div>
+<div data-lang="Python" markdown="1">
+{% highlight python %}
+from pyflink.table import *
+
+settings = EnvironmentSettings.new_instance().in_batch_mode().use_blink_planner().build()
+t_env = StreamTableEnvironment.create(environment_settings=settings)
```
Comment (WeiZhong94) [inline docs/dev/table/hive/hive_dialect.zh.md:91]: BatchTableEnvironment?

File: docs/dev/table/hive/index.md [hunk 5]
```diff
@@ -336,6 +338,25 @@ tableEnv.registerCatalog("myhive", hive)
 tableEnv.useCatalog("myhive")
 {% endhighlight %}
 </div>
+<div data-lang="Python" markdown="1">
+{% highlight python %}
+from pyflink.table import *
+from pyflink.table.catalog import HiveCatalog
+
+settings = EnvironmentSettings.new_instance().in_batch_mode().use_blink_planner().build()
+t_env = StreamTableEnvironment.create(environment_settings=settings)
```
Comment (WeiZhong94) [inline docs/dev/table/hive/index.md:347]: ditto

File: docs/dev/table/hive/index.zh.md [hunk 6]
```diff
@@ -331,6 +333,26 @@ tableEnv.registerCatalog("myhive", hive)
...
 {% endhighlight %}
 </div>
+</div>
+<div data-lang="Python" markdown="1">
+{% highlight python %}
+from pyflink.table import *
+from pyflink.table.catalog import HiveCatalog
+
+settings = EnvironmentSettings.new_instance().in_batch_mode().use_blink_planner().build()
+t_env = StreamTableEnvironment.create(environment_settings=settings)
```
Comment (WeiZhong94) [inline docs/dev/table/hive/index.zh.md:343]: ditto

**Comment by dianfu** [comment]
@WeiZhong94 Thanks for the review. I'll address the comments when merging the PR.

**Pull request closed without merging**